### PR TITLE
Add test code for creating directly chained when cloning job

### PR DESCRIPTION
**Pull request by Amrysliu**
Related: https://github.com/os-autoinst/openQA/pull/2974

### CODE REVIEW

File: t/40-openqa-clone-job.t [hunk 1]
```diff
@@ -21,14 +21,16 @@ use Test::Exception;
 use Test::More;
 use FindBin;
 use lib "$FindBin::Bin/lib";
-use OpenQA::Test::Utils qw(run_cmd test_cmd);
-
+use Date::Format 'time2str';
+use Mojo::File qw(path curfile tempfile);
+use OpenQA::Test::Utils qw(run_cmd test_cmd stop_service);
+use OpenQA::Test::Database;
```
Comment (okurz) [inline t/40-openqa-clone-job.t:27]: I recently saw this inconsistently handled in many files but I would prefer if we can group imports a little better, at least group by "external" and "internal" dependencies. Move every *external* dependency *above* the `use FindBin`, e.g.:

```perl
use Date::Format 'time2str';
use Mojo::File qw(path curfile tempfile);
use FindBin;
use lib "$FindBin::Bin/lib";
use OpenQA::Test::Utils qw(run_cmd test_cmd stop_service);
use OpenQA::Test::Database;
```

as we only need the `use lib…` to resolve all internal modules.

File: t/40-openqa-clone-job.t [hunk 2]
```diff
@@ -21,14 +21,16 @@ use Test::Exception;
...
 
 sub test_once {
     # Report failure at the callsite instead of the test function
     local $Test::Builder::Level = $Test::Builder::Level + 1;
     # prevent all network access to stay local
-    test_cmd('unshare -r -n script/openqa-clone-job', @_);
+    test_cmd(path(curfile->dirname, '../script/openqa-clone-job')->realpath, @_);
```
Comment (okurz) [inline t/40-openqa-clone-job.t:33]: I prefer to still prevent network access by default to prevent us doing accidental harmful remote actions.

**Comment by Amrysliu** [comment]
> Maybe `t/40-openqa-clone-job.t` is not a good place for your test addition.
> 
> Also, the test time increased significantly now. Comparing the current state of master in https://app.circleci.com/pipelines/github/os-autoinst/openQA/2527/workflows/33940c21-2451-43a8-b073-f600b9e2cdd7/jobs/23764
> I find a runtime of `t/40-openqa-clone-job.t ........................ ok 11762 ms` vs. in your case `t/40-openqa-clone-job.t ........................ ok 26307 ms` which I do not see as a worthful increase for the considered change.
> 
After all this test script added more operations e.g. create a test database, rest api call. I compare the runtime with other similar test script's, I do not think it's unnormal. e.g.`t/40-script_load_templates.t ................... ok    38573 ms`
> Have you looked into t/09-job_clone.t ?
> 
I did not look into t/09-job_clone.t until I see this comment.
> Maybe this test module or even another one is a better place, especially when they already create a test database.

**Comment by okurz** [comment]
Yes, other scripts also take a long time. However both overall runtime matters as well as individual tests should not run long. And every test that takes long is frustrating to use for local development.

**Comment by okurz** [comment]
> If I put this test code into 09-job_clone.t, how to test the code in openqa-clone-job

I recommend an approach similar as in https://github.com/os-autoinst/openQA/pull/2985 to move code from the clone script into a lib module, e.g. OpenQA::Scripts::CloneJob after we have https://github.com/os-autoinst/openQA/pull/2985 merged

**Comment by Amrysliu** [comment]
> > If I put this test code into 09-job_clone.t, how to test the code in openqa-clone-job
> 
> I recommend an approach similar as in #2985 to move code from the clone script into a lib module, e.g. OpenQA::Scripts::CloneJob after we have #2985 merged

ok, I got it. I prefer to close this pull request since the clone script will be improved soon.

**Comment by okurz** [comment]
I do not see a reason why you should need to close this PR. Just update it accordingly.

**Pull request closed without merging**